### PR TITLE
added LiquiBase

### LINKED ISSUE #3: Set up Liquibase in the project
Use this guide https://reflectoring.io/database-migration-spring-boot-liquibase/ for setting up Liquibase for the app.

We need table `user` with the following fields:
- id
- first_name
- last_name
- email
- password

### LINKED ISSUE #5: Implement `User` entity
Implement `User` entity with fields according to #3 
Use Lombok for getters, setters, constructors, etc.

#### Useful links
https://www.baeldung.com/lombok-ide#intellij

### PR DESCRIPTION
**Pull request by andy-sam**
Fixes #3
Fixes #5

### CODE REVIEW

File: src/main/resources/db/changelog/changelog-2021-06-18-01.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: username
+                  type: varchar(50)
+                  constraints:
+                    unique: true
+                    nullable: false
```
Comment (alex-anakin) [inline src/main/resources/db/changelog/changelog-2021-06-18-01.yaml:26]: we don't need `username` column, please remove

File: src/main/resources/db/changelog/changelog-2021-06-18-01.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+              - column:
+                  name: first_name
+                  type: varchar(50)
+              - column:
+                  name: last_name
+                  type: varchar(50)
+              - column:
+                  name: email
+                  type: varchar(50)
```
Comment (alex-anakin) [inline src/main/resources/db/changelog/changelog-2021-06-18-01.yaml:40]: please add an empty line

File: src/main/resources/db/changelog/changelog-2021-06-18-01.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                  constraints:
+                    unique: true
+                    nullable: false
+              - column:
+                  name: password
+                  type: varchar(512)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: first_name
```
Comment (alex-anakin) [inline src/main/resources/db/changelog/changelog-2021-06-18-01.yaml:33]: please add a not-null constraint

File: src/main/resources/db/changelog/changelog-2021-06-18-01.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                  type: varchar(50)
+              - column:
+                  name: last_name
```
Comment (alex-anakin) [inline src/main/resources/db/changelog/changelog-2021-06-18-01.yaml:36]: please add a not-null constraint

File: src/main/resources/db/changelog/changelog-2021-06-18-01.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                  type: varchar(50)
+              - column:
+                  name: email
```
Comment (alex-anakin) [inline src/main/resources/db/changelog/changelog-2021-06-18-01.yaml:39]: please add a not-null constraint

File: src/main/java/com/codesarge/banking/User.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package com.codesarge.banking;
+
+import javax.persistence.*;
+import java.io.Serializable;
+
+@Entity
+@Table(name = "user")
+public class User implements Serializable {
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/User.java:8]: use Lombok for getters and setters

File: src/main/java/com/codesarge/banking/User.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private Long id;
+
+    @Column(name = "username", unique = true, nullable = false)
+    private String userName;
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/User.java:16]: please remove this field

File: src/main/java/com/codesarge/banking/User.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Column(name = "password", nullable = false)
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/User.java:18]: please check out, do we need to specify column name if it's equal to the field name

File: src/main/java/com/codesarge/banking/User.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    private String password;
+
+    @Column(name = "first_name")
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/User.java:21]: nullable = false

File: src/main/java/com/codesarge/banking/User.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    private String firstName;
+
+    @Column(name = "last_name")
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/User.java:24]: nullable = false

File: src/main/java/com/codesarge/banking/User.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    private String lastName;
+
+    @Column(name = "email")
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/User.java:27]: nullable = false

File: src/main/java/com/codesarge/banking/User.java [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+import javax.persistence.*;
+import java.io.Serializable;
+
+@Entity
+@Table(name = "user")
+public class User implements Serializable {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/User.java:12]: please check out, do we need to specify column name if it's equal to the field name

**Review by alex-anakin — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties [hunk 13]
```diff
@@ -3,4 +3,6 @@ spring.datasource.url=jdbc:mysql://${MYSQL_HOST:localhost}:3306/banking
 spring.datasource.username=root
 spring.datasource.password=Aa-123456
 spring.datasource.driver-class-name=com.mysql.jdbc.Driver
+
+#spring.liquibase.change-log=classpath:db/changelog/db.changelog-master.yaml
```
[the remark this answers is not in the document]
Comment (andy-sam) [reply src/main/resources/application.properties:7]: done

File: src/main/resources/application.properties [hunk 14]
```diff
@@ -3,4 +3,6 @@ spring.datasource.url=jdbc:mysql://${MYSQL_HOST:localhost}:3306/banking
...
 #spring.jpa.show-sql: true
```
[the remark this answers is not in the document]
Comment (andy-sam) [reply src/main/resources/application.properties:8]: done

File: src/main/resources/db/changelog/changelog-2021-06-18-01.yaml (hunk 1 above)
Comment (andy-sam) [reply src/main/resources/db/changelog/changelog-2021-06-18-01.yaml:26]: done

File: src/main/resources/db/changelog/changelog-2021-06-18-01.yaml (hunk 2 above)
Comment (andy-sam) [reply src/main/resources/db/changelog/changelog-2021-06-18-01.yaml:40]: done

File: src/main/resources/db/changelog/changelog-2021-06-18-01.yaml (hunk 3 above)
Comment (andy-sam) [reply src/main/resources/db/changelog/changelog-2021-06-18-01.yaml:33]: done

File: src/main/java/com/codesarge/banking/User.java (hunk 8 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/User.java:18]: deleted field "username"

File: src/main/java/com/codesarge/banking/User.java (hunk 9 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/User.java:21]: I didn't understand what you mean =(

File: src/main/java/com/codesarge/banking/User.java (hunk 10 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/User.java:24]: done

File: src/main/java/com/codesarge/banking/User.java (hunk 11 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/User.java:27]: done

File: src/main/java/com/codesarge/banking/User.java (hunk 12 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/User.java:12]: column name is deleted

**Review by alex-anakin — APPROVED** [review]

**Pull request merged**